RFS/src/main/java/org/opensearch/migrations/bulkload/lucene/LuceneDocumentsReader.java

```diff
@@ -0,0 +1,39 @@
+package org.opensearch.migrations.bulkload.lucene;
+
+import java.nio.file.Path;
+
+import org.opensearch.migrations.VersionMatchers;
+import org.opensearch.migrations.bulkload.common.RfsLuceneDocument;
+import org.opensearch.migrations.cluster.ClusterSnapshotReader;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Flux;
+
+public interface LuceneDocumentsReader {
+    Flux<RfsLuceneDocument> readDocuments(int startSegmentIndex, int startDoc);
+
+    @Slf4j
+    @AllArgsConstructor
+    public static class Factory {
+        public ClusterSnapshotReader snapshotReader;
+
+        public LuceneDocumentsReader getReader(Path path) {
+            if (VersionMatchers.isES_5_X.or(VersionMatchers.isES_6_X).test(snapshotReader.getVersion())) {
```

Comment:
Unless I'm missing something, we shouldn't use a Lucene 7 reader to read an ES 5 index, because that ES5 index may contain Lucene 5 segments.

Reply:
Good callout, I've updated https://opensearch.atlassian.net/browse/MIGRATIONS-2276 's AC to include adding lucene6 when handing ES5 clusters because we will have the same issue with ES2 -> ES5 upgraded clusters.

Reply:
TBH, I was a bit nervous here but we don't officially support ES 5 sources yet so any issues that arise will be because folks are using RFS off-label.  The task to add official ES 5 support will resolve this gap.  Fine w/ me.